smartsim/entity/dbobject.py

```diff
@@ -107,18 +109,24 @@ def _enumerate_devices(self) -> t.List[str]:
         :rtype: list[str]
         """
         devices = []
-        if ":" in self.device and self.devices_per_node > 1:
-            msg = "Cannot set devices_per_node>1 if a device numeral is specified, "
-            msg += f"the device was set to {self.device} and devices_per_node=={self.devices_per_node}"
-            raise ValueError(msg)
-        if self.device in ["CPU", "GPU"] and self.devices_per_node > 1:
+        if self.device in ["GPU"] and self.devices_per_node > 1:
             for device_num in range(self.devices_per_node):
                 devices.append(f"{self.device}:{str(device_num)}")
         else:
             devices = [self.device]
 
         return devices
 
+    def _check_arguments(self):
+        devices = []
+        if ":" in self.device and self.devices_per_node > 1:
```

Comment:
if we do a check at the start:
```
if self.devices_per_node <= 1:
  return
```

We can avoid having compound conditionals below, making it easier to read what's required to trigger a failure.